sympy/sets/tests/test_setexpr.py

```diff
@@ -89,97 +89,116 @@ def test_Interval_arithmetic():
     assert n23cc + n23ro == SetExpr(Interval.Ropen(-4, 6))
     assert n23cc - n23ro == SetExpr(Interval.Lopen(-5, 5))
-    assert n23cc * n23ro == SetExpr(Interval.Ropen(-6, 9))
-    assert n23cc / n23ro == SetExpr(Interval.Lopen(-oo, oo))
+    assert n23cc*n23ro == SetExpr(Interval.Ropen(-6, 9))
+    assert n23cc/n23ro == SetExpr(Interval.Lopen(-oo, oo))
     assert n23cc + n23lo == SetExpr(Interval.Lopen(-4, 6))
     assert n23cc - n23lo == SetExpr(Interval.Ropen(-5, 5))
-    assert n23cc * n23lo == SetExpr(Interval(-6, 9))
-    assert n23cc / n23lo == SetExpr(Interval.open(-oo, oo))
+    assert n23cc*n23lo == SetExpr(Interval(-6, 9))
+    assert n23cc/n23lo == SetExpr(Interval.open(-oo, oo))
     assert n23cc + n23o == SetExpr(Interval.open(-4, 6))
     assert n23cc - n23o == SetExpr(Interval.open(-5, 5))
-    assert n23cc * n23o == SetExpr(Interval.open(-6, 9))
-    assert n23cc / n23o == SetExpr(Interval.open(-oo, oo))
-    assert n23cc ** 2 == SetExpr(Interval(0, 9))
-    assert n23cc ** 3 == SetExpr(Interval(-8, 27))
+    assert n23cc*n23o == SetExpr(Interval.open(-6, 9))
+    assert n23cc/n23o == SetExpr(Interval.open(-oo, oo))
+    assert n23cc**2 == SetExpr(Interval(0, 9))
+    assert n23cc**3 == SetExpr(Interval(-8, 27))
 
     n32cc = SetExpr(Interval(-3, 2))
     n32lo = SetExpr(Interval.Lopen(-3, 2))
     n32ro = SetExpr(Interval.Ropen(-3, 2))
-    assert n32cc * n32lo == SetExpr(Interval.Ropen(-6, 9))
-    assert n32cc * n32cc == SetExpr(Interval(-6, 9))
-    assert n32lo * n32cc == SetExpr(Interval.Ropen(-6, 9))
-    assert n32cc * n32ro == SetExpr(Interval(-6, 9))
-    assert n32lo * n32ro == SetExpr(Interval.Ropen(-6, 9))
-    assert n32cc / n32lo == SetExpr(Interval.Ropen(-oo, oo))
-    assert i12cc / n32lo == SetExpr(Interval.Ropen(-oo, oo))
+    assert n32cc*n32lo == SetExpr(Interval.Ropen(-6, 9))
+    assert n32cc*n32cc == SetExpr(Interval(-6, 9))
+    assert n32lo*n32cc == SetExpr(Interval.Ropen(-6, 9))
+    assert n32cc*n32ro == SetExpr(Interval(-6, 9))
+    assert n32lo*n32ro == SetExpr(Interval.Ropen(-6, 9))
+    assert n32cc/n32lo == SetExpr(Interval.Ropen(-oo, oo))
+    assert i12cc/n32lo == SetExpr(Interval.Ropen(-oo, oo))
 
-    assert n3n2cc ** 2 == SetExpr(Interval(4, 9))
-    assert n3n2cc ** 3 == SetExpr(Interval(-27, -8))
+    assert n3n2cc**2 == SetExpr(Interval(4, 9))
+    assert n3n2cc**3 == SetExpr(Interval(-27, -8))
 
     assert n23cc + i12cc == SetExpr(Interval(-1, 5))
     assert n23cc - i12cc == SetExpr(Interval(-4, 2))
-    assert n23cc * i12cc == SetExpr(Interval(-4, 6))
-    assert n23cc / i12cc == SetExpr(Interval(-2, 3))
+    assert n23cc*i12cc == SetExpr(Interval(-4, 6))
+    assert n23cc/i12cc == SetExpr(Interval(-2, 3))
+
+
+def test_Interval_inf():
+    neinf = SetExpr(-oo)
+    poinf = SetExpr(oo)
+    set_ff = SetExpr(Interval(0, 5))
+    set_if = SetExpr(Interval(-oo, 5))
+    set_fi = SetExpr(Interval(0, oo))
+    set_ii = SetExpr(Interval(-oo, oo))
+
+    assert set_ff + neinf == set_if
```

Comment:
`set_ff + neinf` should be `neinf`.

Reply:
@jksuom , could you please give me a mathematical explanation about this?